samples/WebApp/JavaScript.html

```diff
@@ -0,0 +1,109 @@
+<!--
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ -->
+<script>
+  /**
+   * Create a wrapped version of google.script.run that
+   * is adapted for Angular promises.
+   */
+  var ScriptService = function($q) {
+    var self = this;
+    var promisify = function(key) {
+      return function() {
+        var args = arguments;
+        return $q(function(resolve, reject) {
+          google.script.run
+            .withSuccessHandler(resolve)
+            .withFailureHandler(reject)
+            [key].apply(google.script.run, args);
+        });
+      };
+    };
+    angular.forEach(google.script.run, function(_, key) {
+      self[key] = promisify(key)
+    });
+  };
+
+  /**
+   * The controller for the sample.
+   */
+  var SampleController = function($scope, $window, script, intercom) {
+    var self = this;
+
+    // Hold onto some dependencies.
+    self.$window = $window;
+    self.script = script;
+
+    // Initialize state.
+    self.email = $window.email;
+    self.isSignedIn = $window.isSignedIn;
+    self.user = null;
+    self.repos = null;
+
+    // Watch for changes to isSignedIn.
+    $scope.$watch('sample.isSignedIn', function(isSignedIn) {
```

Comment:
For me it would be helpful to add a link as to what this means.
https://docs.angularjs.org/api/ng/type/$rootScope.Scope#$watch

> Watch for changes to isSignedIn.

This statement can be inferred from the next line. But what does the block do? We watch for changes and then... (get a GitHub profile and user?)

Reply:
Expanded on the comment.